apps/web/components/booking/CancelBooking.tsx

```diff
@@ -45,6 +45,17 @@ export default function CancelBooking(props: Props) {
   const telemetry = useTelemetry();
   const [error, setError] = useState<string | null>(booking ? null : t("booking_already_cancelled"));
 
+  const { eventType } = bookingCancelledEventProps;
+  let isBookingCancellationReasonRequired = false;
+  if (
+    eventType &&
+    typeof eventType === "object" &&
+    "bookingCancellationReasonRequired" in eventType &&
+    typeof eventType.bookingCancellationReasonRequired === "boolean"
+  ) {
+    isBookingCancellationReasonRequired = eventType.bookingCancellationReasonRequired;
+  }
```

Comment:
This is not needed.` bookingCancellationReasonRequired` is not an optional property; it's either true or false. We can use it directly. Currently, we use unknown as the type for the event and booking props in this component. If we fix that, we won't need to do any extra work. eg:- `eventType: PageProps["eventType"]`;

Reply:
I think eventtype is set as unknown for some reason. Not completely sure about it which is why I did not change it.

cc @Udit-takkar